### PR TITLE
feat(pymongo): add support for v4.9

### PR DESCRIPTION
**Pull request by mabdinur**
## Motivation

- Add and test support for pymongo v4.9.1

## Reproduction

```
pip install ddtrace==2.11.6 pymongo==4.9.1
python -c "import ddtrace; ddtrace.patch_all(pymongo=True); import pymongo"
```

## Output

```
  File "dd-trace-py/ddtrace/_monkey.py", line 165, in on_import
    imported_module = importlib.import_module(path)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/munirabdinur/.pyenv/versions/3.12.2/lib/python3.12/importlib/__init__.py", line 90, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen importlib._bootstrap>", line 1387, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1360, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1331, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 935, in _load_unlocked
  File "dd-trace-py/ddtrace/internal/module.py", line 309, in _exec_module
    self.loader.exec_module(module)
  File "<frozen importlib._bootstrap_external>", line 995, in exec_module
  File "<frozen importlib._bootstrap>", line 488, in _call_with_frames_removed
  File "dd-trace-py/ddtrace/contrib/pymongo/__init__.py", line 45, in <module>
    from .patch import get_version
  File "<frozen importlib._bootstrap>", line 1360, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1331, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 935, in _load_unlocked
  File "dd-trace-py/ddtrace/internal/module.py", line 309, in _exec_module
    self.loader.exec_module(module)
  File "dd-trace-py/ddtrace/contrib/pymongo/patch.py", line 51, in <module>
    _VERIFY_VERSION_CLASS = pymongo.pool.SocketInfo if _VERSION < (4, 5) else pymongo.pool.Connection
```

## Checklist
- [x] PR author has checked that all the criteria below are met
- The PR description includes an overview of the change
- The PR description articulates the motivation for the change
- The change includes tests OR the PR description describes a testing strategy
- The PR description notes risks associated with the change, if any
- Newly-added code is easy to change
- The change follows the [library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html)
- The change includes or references documentation updates if necessary
- Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))

## Reviewer Checklist
- [x] Reviewer has checked that all the criteria below are met 
- Title is accurate
- All changes are related to the pull request's stated goal
- Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes
- Testing strategy adequately addresses listed risks
- Newly-added code is easy to change
- Release note makes sense to a user of the library
- If necessary, author has acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment
- Backport labels are set in a manner that is consistent with the [release branch maintenance policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)

### CODE REVIEW

File: releasenotes/notes/pymong-4-9-support-83f7c613e5e009e6.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+---
+features:
+  - |
+    pymongo: Adds support for pymongo>=4.9.0
```
Comment (mabdinur) [inline releasenotes/notes/pymong-4-9-support-83f7c613e5e009e6.yaml:4]:
```suggestion
    pymongo: Adds support for pymongo>=4.9.0

```

**Comment by mabdinur** [comment]
Since pymongo is a popular library it would be nice if we could sneak this change in to the v2.13.0 release @erikayasuda 🙏

**Comment by erikayasuda** [comment]
> Since pymongo is a popular library it would be nice if we could sneak this change in to the v2.13.0 release @erikayasuda 🙏

I can get back to you on this 🫡

File: riotfile.py [hunk 2]
```diff
@@ -1434,7 +1434,8 @@ def select_pys(min_version=MIN_PYTHON_VERSION, max_version=MAX_PYTHON_VERSION):
             name="mongoengine",
             command="pytest {cmdargs} tests/contrib/mongoengine",
             pkgs={
-                "pymongo": latest,
```
Comment (erikayasuda) [inline riotfile.py:1437]: [Line 749] Should we be testing the specific versions you have if/else for in the patch code? Specifically 4.5 (maybe 4.9 but I guess that's a bit overkill with `latest` already there)

**Review by erikayasuda — APPROVED** [review]
Just a nit, otherwise LGTM

File: riotfile.py (hunk 2 above)
Comment (mabdinur) [reply riotfile.py:1437]: We already do this for pymongo. Our mongoengine essentially just enables our pymongo integration. It doesn't do anything special. 

I am actually planning to remove the mongoengine integration in v3.0.0

File: ddtrace/contrib/internal/pymongo/patch.py [hunk 3]
```diff
@@ -60,43 +73,39 @@ def unpatch():
 
 def patch_pymongo_module():
     _w(pymongo.MongoClient.__init__, _trace_mongo_client_init)
-    _w(pymongo.topology.Topology.select_server, _trace_topology_select_server)
+    _w(Topology.select_server, _trace_topology_select_server)
```
Comment (quinna-h) [inline ddtrace/contrib/internal/pymongo/patch.py:76]: For testing do we just want to assert that we've successfully wrapped these methods?

Comment (mabdinur) [reply ddtrace/contrib/internal/pymongo/patch.py:76]: Yup

**Pull request merged**